syncserver/src/tokenserver/extractors.rs

```diff
@@ -98,14 +98,6 @@ impl TokenserverRequest {
             });
         }
 
-        // If the client previously reported a client state, every subsequent request must include
-        // one. Note that this is only relevant for BrowserID requests, since OAuth requests must
-        // always include a client state.
-        if !self.user.client_state.is_empty() && self.auth_data.client_state.is_empty() {
```

Comment:
Double checking these docs stating these sections are "only relevant for BrowserID", I'm not entirely sure about this one -- where do we enforce a non empty `client_state` for oauth? It's a bit more complicated by the fact that we check different places for it (via `X-KeyID` and `X-Client-State`) headers.

I probably need to stare at this code a little more closely but I'm also wondering if a test for this case might be helpful (if it doesn't exist already) -- could be a separate issue.

Reply:
meh, leaving the code and removing the comment seems reasonable.